lib/actions/apiV2.js

```diff
@@ -651,7 +652,33 @@ export const findRoute = (params) =>
 
             const newRoute = clone(route)
             const routePatterns = {}
-            newRoute.patterns.forEach((pattern) => {
+
+            // Sort patterns by length to make algorithm below more efficient
+            const patternsSortedByLength = newRoute.patterns.sort(
+              (a, b) => a.stops.length - b.stops.length
+            )
+
+            // Remove all patterns that are subsets of larger patterns
+            const filteredPatterns = patternsSortedByLength
+              // Start with the largest for performance
+              .reverse()
+              .filter((pattern) => {
+                // Compare to all other patterns TODO: make this beat O(n^2)
+                return !patternsSortedByLength.find((p) => {
+                  // Don't compare against ourself
+                  if (p.id === pattern.id) return false
+
+                  // If our pattern is longer, it's not a subset
+                  if (p.stops.length < pattern.stops.length) return false
+
+                  return isValidSubsequence(
+                    p.stops.map((s) => s.id),
+                    pattern.stops.map((s) => s.id)
+                  )
+                })
+              })
```

Comment:
I think we need to check the pattern id because the indexes don't always line up. I agree the length check is not required but to do that we need to start checking the second array at the right index and that logic is currently not present